torch/csrc/distributed/c10d/init.cpp

```diff
@@ -2316,6 +2316,12 @@ options :class:`~torch.distributed.ProcessGroupNCCL.Options`).
               },
               py::arg("abort_reason") = py::none(),
               py::call_guard<py::gil_scoped_release>())
+          .def(
+              "_close",
```

Comment:
There's already a `abort` method exposed in ProcessGroup API, why introduce yet another API and what's the difference between these APIs?

Reply:
`_abort` only aborts the PG, `_close` does an abort and also cleans up the background threads. `_close` is a superset of `_abort` and for our usecase, we can just use `_close` instead. See `ProcessGroupNCCL::close()` vs `ProcessGroupNCCL::abort()`.

I can remove `_abort` if no one is using it. I thought I'd keep it since maybe someone else might be using it.

Reply:
`_abort` is a private API and we don't have doc on it. I feel we should just remove it so that no new ppl rely on it. I would prefer we avoid adding a new API with such similar functionality. For naming, I think either `_close` or `_shutdown` looks good to me.

Reply:
I vote for the name `_shutdown` or `_force_shutdown` because this looks like a user side action instead of a natural event by the code?

Reply:
Renamed to `_shutdown` and removed `_abort`